### PR TITLE
chore(broker-core): remove unnecessary #runInTransaction

### LINKED ISSUE #2138: Remove nested transaction calls
After migrating to a transactional DB, we still have some calls leftover (e.g. in the `JobState`) which want to execute some things in a transaction, even though the whole process record runs in a transaction.

This isn't urgent, as we're not actually running nested transactions (even if it looks like it), but we should clean it up anyway.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #2138

### CODE REVIEW

**Comment by MiguelPires** [comment]
IIRC, the decision to keep these transaction calls was a conscious one. The reasoning was that from the perspective of someone who's writing code in one the *State classes, it's easier and less error prone if you do it in a transaction when you need it instead of having to think about whether it's already running in the ProcessingStateMachine. But I don't know if you discussed this in the meantime and the consensus has changed (if so, please disregard what I just wrote)

**Comment by Zelldon** [comment]
Yeah would be nice if the decision have been documented somewhere.
For me this argument is a bit unfounded, because we will always run in the StreamProcessController except in *one* case. The thing is that if we use the nested transaction we also nest the catched exceptions, which will a bit weird.

File: broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageSubscriptionState.java [hunk 1]
```diff
@@ -149,21 +146,18 @@ public void resetSentTime(MessageSubscription subscription) {
   }
 
   public void updateSentTime(final MessageSubscription subscription, long sentTime) {
-    dbContext.runInTransaction(
```
Comment (MiguelPires) [inline broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageSubscriptionState.java]: This and visitSubscriptions are called by the PendingMessageSubscriptionChecker which doesn't seem to be running in a transaction, although I might be understanding it incorrectly.

File: broker-core/src/main/java/io/zeebe/broker/subscription/message/state/WorkflowInstanceSubscriptionState.java [hunk 2]
```diff
@@ -133,6 +130,7 @@ public void updateToClosingState(final WorkflowInstanceSubscription subscription
   }
 
   public void updateSentTime(final WorkflowInstanceSubscription subscription, long sentTime) {
+    // this is called from an pending checker which needs a transaction
```
Comment (MiguelPires) [inline broker-core/src/main/java/io/zeebe/broker/subscription/message/state/WorkflowInstanceSubscriptionState.java]: Wouldn't it be better to put this transaction in the pending checker? Here you don't know whether you're in a transaction or not. This is also called from other places like the above update methods which are called by processors so there's the issue you mentioned with the nested transactions. You'd have to expose the dbContext though, so I'm not sure if it's that much better

**Review by miguelpires — CHANGES_REQUESTED** [review]
I agree with your point about the nested transactions. We could also check in runInTransaction if the exception is a ZeebeDbException, so we don't risk rethrowing it incorrectly if there are nested transactions. In this PR, I only pointed out two small things

File: broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageSubscriptionState.java (hunk 1 above)
Comment (Zelldon) [reply broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageSubscriptionState.java]: True for update sentTime I still have the transaction and for the visit internally we always ensure that is run in a transaction.

File: broker-core/src/main/java/io/zeebe/broker/subscription/message/state/WorkflowInstanceSubscriptionState.java (hunk 2 above)
Comment (Zelldon) [reply broker-core/src/main/java/io/zeebe/broker/subscription/message/state/WorkflowInstanceSubscriptionState.java]: You're right.

**Comment by Zelldon** [comment]
> I agree with your point about the nested transactions. We could also check in runInTransaction if the exception is a ZeebeDbException, so we don't risk rethrowing it incorrectly if there are nested transactions.

Yes I will do this, but I thing it makes sense to have an extra issue for this right? Or should I just fix this here?

**Comment by MiguelPires** [comment]
Yeah, I think it makes sense to have that in a separate issue

**Comment by Zelldon** [comment]
Created an issue #2281 and added new commits to apply the review hints

**Comment by npepinpe** [comment]
I was wondering, since those are state changes applied outside of the stream processor controller, could they somehow mess with the snapshots? I think in this case it doesn't really matter, but if we do state changes somewhere else outside of stream processing, could it be a problem? Applying changes that get snapshotted but the related command is not replicated, for example. Kind of an edge case anyway, I'm just wondering if it could be an issue.

**Comment by Zelldon** [comment]
Since this runs in the same actor, which means same thread this is not an issue I would argue.

**Comment by npepinpe** [comment]
Yes, you're right, and the snapshot wouldn't be valid if the event is not committed, was just wondering

**Review by miguelpires — APPROVED** [review]

**Pull request merged**